src/debug/createdump/crashinfo.cpp

```diff
@@ -439,13 +439,21 @@ CrashInfo::GetDSOInfo()
         if (!ReadMemory(linkMapAddr, &map, sizeof(map))) {
             return false;
         }
-        char moduleName[257] = { 0 };
+        ArrayHolder<char> moduleName = new char[MAX_LONGPATH];
+        moduleName[0] = '\0';
         if (map.l_name != nullptr) {
-            if (!ReadMemory(map.l_name, &moduleName, sizeof(moduleName) - 1)) {
-                return false;
+            for (int i = 0; i < MAX_LONGPATH; i++)
+            {
+                if (!ReadMemory(map.l_name + i, &moduleName[i], 1)) {
+                    TRACE("DSO: ReadMemory link_map name %p + %d FAILED\n", map.l_name, i);
+                    break;
+                }
+                if (moduleName[i] == '\0') {
+                    break;
+                }
             }
         }
-        TRACE("DSO: link_map entry %p l_ld %p l_addr %lx %s\n", linkMapAddr, map.l_ld, map.l_addr, moduleName);
+        TRACE("DSO: link_map entry %p l_ld %p l_addr %lx %s\n", linkMapAddr, map.l_ld, map.l_addr, (char*)moduleName);
```

Comment:
You probably want to set moduleName[MAX_LONGPATH-1] = 0 as a precaution. Otherwise, looks good.